### DISCUSSION TITLE
[RTC] LiveKit RTC Implementation

### DISCUSSION THREAD

**Discussion by cyfyifanchen** [TEN-framework/ten-framework#288] [General]
We've got quite a few folks asking how to implement or replace Agora RTC with LiveKit RTC, one gentleman mentioned it's for self-hosted purpose. 

We might want to evaluate the pros and cons of it.

**Comment by Zuckonit** [comment]
good idea to make rtc pluggable

**Comment by remisharrock** [comment]
same question here

**Reply by cyfyifanchen** [reply]
@remisharrock a man smart like you, can totally contribute a Livekit ext. 👀

**Reply by remisharrock** [reply]
Absolutely 😜 
Pipecat has for example multiple implementations of the transport layer, not only RTC :
- [webrtc with daily](https://docs.pipecat.ai/server/services/transport/daily)
- [SmallWebRTCTransport](https://docs.pipecat.ai/server/services/transport/small-webrtc)
- [FastAPI WebSocket](https://docs.pipecat.ai/server/services/transport/fastapi-websocket)
- [Standard Websockets](https://docs.pipecat.ai/server/services/transport/websocket-server)

It also has another layer uniformazing everything, that I find very interesting: it's called [RTVI](https://docs.pipecat.ai/server/frameworks/rtvi/introduction) (Real-Time Voice Interaction) and also implementations of rtvi processor and observer.

**Reply by cyfyifanchen** [reply]
@remisharrock we just released TMAN Designer, a so-called visual workflow builder, go give a try. I bet you are gonna like it.
